libcxx/include/optional

```diff
@@ -234,15 +234,18 @@ struct __optional_destruct_base<_Tp, false>
     typedef _Tp value_type;
     static_assert(is_object_v<value_type>,
         "instantiation of optional with a non-object type is undefined behavior");
-#ifdef __CHEERP__
-    struct
-#else
+#if !defined(__CHEERP__) || defined(__ASMJS__)
     union
-#endif
     {
+#endif
         char __null_state_;
-        value_type __val_;
+        value_type __val_
+#if !defined(__CHEERP__) || defined(__ASMJS__)
+        ;
     };
+#else
+        [[cheerp::noinit]];
```

Comment:
Can we use it unconditionally? In the case of unions it should have no effect anyway

Reply:
not so easy, because if we don't have the union there is no item to which to attach the attribute

Reply:
Not sure. The attribute applies to `__val_`, right?

Reply:
@alexp-sssup should be better now